tests/cpp_unit_tests/test_file_utils.cpp

```diff
@@ -0,0 +1,28 @@
+#include "openmc/file_utils.h"
+#include <catch2/catch_test_macros.hpp>
+
+using namespace openmc;
+
+TEST_CASE("Test get_file_extension")
+{
+  REQUIRE(get_file_extension("rememberthealamo.png") == "png");
+  REQUIRE(get_file_extension("statepoint.20.h5") == "h5");
+  REQUIRE(get_file_extension("wEiRDNaa_ame.h4") == "h4");
+  REQUIRE(get_file_extension("has_directory/asdf.20.h5") == "h5");
+  REQUIRE(get_file_extension("wasssssup_lol") == "");
+  REQUIRE(get_file_extension("has_directory/secret_file") == "");
```

Comment:
Maybe one more you could add is "lovely.dir/extensionless_file"

Reply:
very good catch! It was not correct in that case. Should be fixed now.

Reply:
testing ftw :smile: